src/test/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClientTest.java

```diff
@@ -135,154 +175,163 @@ public void testWhenSendMessageWithSetMessageSizeThresholdThenThresholdIsHonored
+        String messageBody = generateStringWithLength(MORE_THAN_SQS_SIZE_LIMIT);
+        SendMessageRequest messageRequest = SendMessageRequest.builder().queueUrl(SQS_QUEUE_URL).messageBody(messageBody).build();
         extendedSqsWithGenericReservedAttributeName.sendMessage(messageRequest);
 
         ArgumentCaptor<SendMessageRequest> sendMessageRequestCaptor = ArgumentCaptor.forClass(SendMessageRequest.class);
         verify(mockSqsBackend).sendMessage(sendMessageRequestCaptor.capture());
 
-        Map<String, MessageAttributeValue> attributes = sendMessageRequestCaptor.getValue().getMessageAttributes();
-        Assert.assertTrue(attributes.keySet().contains(SQSExtendedClientConstants.RESERVED_ATTRIBUTE_NAME));
-        Assert.assertFalse(attributes.keySet().contains(AmazonSQSExtendedClient.LEGACY_RESERVED_ATTRIBUTE_NAME));
+        Map<String, MessageAttributeValue> attributes = sendMessageRequestCaptor.getValue().messageAttributes();
+        Assert.assertTrue(attributes.containsKey(SQSExtendedClientConstants.RESERVED_ATTRIBUTE_NAME));
+        Assert.assertFalse(attributes.containsKey(AmazonSQSExtendedClient.LEGACY_RESERVED_ATTRIBUTE_NAME));
     }
 
     @Test
     public void testWhenSendSmallMessageThenS3IsNotUsed() {
-        int messageLength = SQS_SIZE_LIMIT;
-        String messageBody = generateStringWithLength(messageLength);
+        String messageBody = generateStringWithLength(SQS_SIZE_LIMIT);
 
-        SendMessageRequest messageRequest = new SendMessageRequest(SQS_QUEUE_URL, messageBody);
+        SendMessageRequest messageRequest = SendMessageRequest.builder().queueUrl(SQS_QUEUE_URL).messageBody(messageBody).build();
         extendedSqsWithDefaultConfig.sendMessage(messageRequest);
 
-        verify(mockS3, never()).putObject(isA(PutObjectRequest.class));
+        verify(mockS3, never()).putObject(isA(PutObjectRequest.class), isA(RequestBody.class));
     }
 
     @Test
     public void testWhenSendMessageWithLargePayloadSupportDisabledThenS3IsNotUsedAndSqsBackendIsResponsibleToFailIt() {
-        int messageLength = MORE_THAN_SQS_SIZE_LIMIT;
-        String messageBody = generateStringWithLength(messageLength);
+        String messageBody = generateStringWithLength(MORE_THAN_SQS_SIZE_LIMIT);
         ExtendedClientConfiguration extendedClientConfiguration = new ExtendedClientConfiguration()
                 .withPayloadSupportDisabled();
-        AmazonSQS sqsExtended = spy(new AmazonSQSExtendedClient(mockSqsBackend, extendedClientConfiguration));
-
-        SendMessageRequest messageRequest = new SendMessageRequest(SQS_QUEUE_URL, messageBody);
+        SqsClient sqsExtended = spy(new AmazonSQSExtendedClient(mockSqsBackend, extendedClientConfiguration));
+
+        SendMessageRequest messageRequest = SendMessageRequest.builder()
+            .queueUrl(SQS_QUEUE_URL)
+            .messageBody(messageBody)
+            .overrideConfiguration(
+                AwsRequestOverrideConfiguration.builder()
```

Comment:
can we move this overrider to a helper method so that we can reuse it in other places?